bin/workflows/pycbc_make_inference_inj_workflow

```diff
@@ -0,0 +1,284 @@
+                    help="Path to output directory.")
+parser.add_argument("--output-map", default="output.map",
+                    help="Path to output map file.")
+parser.add_argument("--output-file", required=True,
+                    help="Path to DAX file.")
+
+# input configuration file options
+parser.add_argument("--inference-config-file", type=str, required=True,
+                    help="workflow.WorkflowConfigParser parsable file with "
+                         "proir information.")
+parser.add_argument("--prior-section", type=str, default="prior",
+                    help="Name of the section in inference configuration file "
+                         "that contains priors.")
+
+# input data options
+parser.add_argument("--data-type",
+                    choices=["analytical", "simulated_data"],
+                    default="simulated_data")
+
+# add option groups
+wf.add_workflow_command_line_group(parser)
+
+# parser command line
+opts = parser.parse_args()
+
+# log to terminal until we know where the path to log output file
+log_format = "%(asctime)s:%(levelname)s : %(message)s"
+logging.basicConfig(format=log_format, level=logging.INFO)
+
+# create workflow and sub-workflows
+container = wf.Workflow(opts, opts.workflow_name)
+workflow = wf.Workflow(opts, opts.workflow_name + "-main")
+finalize_workflow = wf.Workflow(opts, opts.workflow_name + "-finalization")
+
+# sections for output HTML pages
+rdir = layout.SectionNumber("results",
+                            ["posteriors", "priors", "workflow"])
+
+# make data output and results directories
+wf.makedir(opts.output_dir)
+wf.makedir(rdir.base)
+wf.makedir(rdir["workflow"])
+
+# create files for workflow log
+log_file_txt = wf.File(workflow.ifos, "workflow-log", workflow.analysis_time,
+                      extension=".txt", directory=rdir["workflow"])
+log_file_html = wf.File(workflow.ifos, "WORKFLOW-LOG", workflow.analysis_time,
+                        extension=".html", directory=rdir["workflow"])
+
+# switch saving log to file
+logging.basicConfig(format=log_format, level=logging.INFO,
+                    filename=log_file_txt.storage_path, filemode="w")
+log_file = logging.FileHandler(filename=log_file_txt.storage_path, mode="w")
+log_file.setLevel(logging.INFO)
+formatter = logging.Formatter(log_format)
+log_file.setFormatter(formatter)
+logging.getLogger("").addHandler(log_file)
+logging.info("Created log file %s" % log_file_txt.storage_path)
+
+# typecast str from command line to File instances
+config_file = to_file(opts.inference_config_file)
+
+# construct Executable for creating injections
+create_injections_exe = wf.Executable(workflow.cp, "create_injections",
+                                      ifos=workflow.ifos,
+                                      out_dir=opts.output_dir)
+
+# construct Executable for running sampler
+inference_exe = wf.Executable(workflow.cp, "inference", ifos=workflow.ifos,
+                              out_dir=opts.output_dir)
```

Comment:
Would it be good to move both of these Executable classes to sub-classes within workflow? If they would be used in other workflows it might be useful.